### PR TITLE
[SERVICES-220] Handlers timing middleware

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
This introduces additional middleware to each handler for performance monitoring. Based on feedback from: https://github.com/Wikia/helios/pull/36 

For now, no changes were included with preparing response JSON body as we should solve that properly as separate task.

/cc @Wikia/services-team

### CODE REVIEW

File: controllers/register_controller.go [hunk 1]
```diff
@@ -57,12 +55,20 @@ func NewRegisterController(
...
 	controller.disableRegistration = serverConfig.DisableRegistration
 
-	http.HandleFunc("/register", originHandler(controller.registerHandler))
-	http.HandleFunc("/username/validation", originHandler(controller.usernameValidationHandler))
+	http.HandleFunc("/register",
+		timedHandler(controller, "registerHandler",
+			originHandler(controller.registerHandler)))
+	http.HandleFunc("/username/validation",
+		timedHandler(controller, "usernameValidationHandler",
+			originHandler(controller.usernameValidationHandler)))
```
Comment (drsnyder) [inline controllers/register_controller.go]: Minor: Does it make sense to combine these into one helper function e.g.:

```go
http.HandleFunc("/register", timedOriginHandler(controller, "registerHandler", controller.registerHandler))
```

**Comment by drsnyder** [comment]
Other than minor comment, LGTM

**Comment by ArturKlajnerok** [comment]
@drsnyder yeah, I think there's not so much logic inside so I've merge that helpers.

**Comment by nmonterroso** [comment]
LGTM

**Pull request merged**